src/main/java/com/algosenpai/app/UserStats.java

```diff
@@ -0,0 +1,175 @@
+ * searched by name.
+ * characterImagePath: a string containing the path of the image that is used for the user's character in game.
+ * The two data above are stored permanently to a text file, and retrieved from the text file when the program starts.
+ *</p>
+ *
+ * <p>
+ * currentChapter: A single ChapterStat object that stores the chapter data of the chapter that is currently running.
+ * This is reset for every run. The contents of currentChapter can be transferred to chapterData by calling
+ * saveCurrentChapterToChapterData()
+ * </p>
+ */
+public class UserStats {
+
+    //username of the player
+    private String username;
+    // Array of chapter stats
+    private ArrayList<ChapterStat> chapterData;
+
+    // Stats for the current chapter
+    private ChapterStat currentChapter;
+
+    private String characterImagePath = "miku.png";
+
+    //Maps the chapter names to an index value
+    private HashMap<String, Integer> chapterNumber;
+
+    /**
+     * Constructs a new UserStats by reading in from the UserData text file.
+     * ChapterStat objects are passed from the parser into here to be stored into
+     * their respective data structures.
+     */
+    public UserStats() throws FileNotFoundException {
+        chapterData = new ArrayList<>();
+        chapterNumber = new HashMap<>();
+        UserStorageParser userStorageParser = new UserStorageParser();
+        //Reads in redundant blank lines
+        userStorageParser.nextLine();
+        userStorageParser.nextLine();
+        this.username = userStorageParser.nextLine();
+
+        while (userStorageParser.hasMoreTokens()) {
+            userStorageParser.nextLine();
+            ChapterStat currStat = userStorageParser.nextChapterStat();
+            chapterData.add(currStat);
+            chapterNumber.put(currStat.chapterName, currStat.chapterNumber);
+        }
+
+    }
+
+    /**
+     * Get the stats for a particular chapter by searching for the chapter number then
+     * calling the getStatsByIndex function.
+     * @param chapterName The name of the chapter.
+     * @return The ChapterStat object for that chapter, or null of the chapter does not exist.
+     */
+    public ChapterStat getStatsByName(String chapterName) {
+        if (chapterNumber.containsKey(chapterName)) {
+            int index = chapterNumber.get(chapterName);
+            return getStatsByIndex(index);
+        }
+        return null;
+    }
+
+    /**
+     * Get the stats for a particular chapter.
+     * @param index The index of the chapter.
+     * @return The ChapterStat object for that chapter
+     */
+    public ChapterStat getStatsByIndex(int index) {
+        ChapterStat currentChapter = chapterData.get(index);
+        return new ChapterStat(currentChapter);
+    }
+
+    /**
+     * Update the stats for the current Chapter according to whether the user answered correctly or incorrectly.
```

Comment:
Starting keyword of a method javadoc comment should be a singular verb `Updates`.